OpenRA.Mods.Common/Util.cs

```diff
@@ -155,5 +158,64 @@ public static IEnumerable<CPos> RandomWalk(CPos p, MersenneTwister r)
 				yield return p;
 			}
 		}
+
+		public static string FriendlyTypeName(Type t)
+		{
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(HashSet<>))
+				return "Set of {0}".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+				return "{0}, {1} pairs".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
```

Comment:
You should be more specific here: (X,Y) pairs implies that duplicate keys with different values would be acceptable.